api-reference/10 UI Widgets/dxDataGrid/1 Configuration/scrolling/mode.md

```diff
@@ -12,15 +12,15 @@ Specifies the scrolling mode.
 
 - **Standard**      
-A user scrolls a single page only. Scrolling appears only if [all the page's rows](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize') do not fit into the widget's height. In this mode, the [pager](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If [paging](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/') is disabled, the widget loads all rows at once, which may decrease the widget's performance. In this case, we recommend choosing another scrolling mode.
-
-- **Virtual**       
-This mode is an alternative to paging where pages are loaded when they get into the viewport and removed once they leave it. Use this mode if a user should be able to scroll data by pages.
+A user scrolls a single page only. Scrolling is available only if [all the page's rows](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/#pageSize') do not fit into the widget's height. In this mode, the [pager](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/pager '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/pager/') performs the main navigation and scrolling is auxiliary. If [paging](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/paging/') is disabled, the widget loads all rows simultaneously which can reduce the widget's performance. In this case, we recommend that you use another scrolling mode.
 
 - **Infinite**      
-This mode is an alternative to paging where each next page is loaded once the scrollbar reaches the end of its scale. Use this mode if a user should scroll data gradually, from the first to the last page.
+Each next page is loaded once the scrollbar reaches the end of its scale. In this mode, users scroll data gradually, from the first to the last page.
 
     [note]Set the **grouping**.[allowCollapsing](/api-reference/10%20UI%20Widgets/dxDataGrid/1%20Configuration/grouping/allowCollapsing.md '/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/grouping/#allowCollapsing') option to **false** when using infinite scrolling in conjunction with grouping.
+    
+- **Virtual**       
+Rows are loaded when they get into the viewport and removed once they leave it. If the rows take time to be loaded and rendered, they display gray boxes. [Rendering optimization](/Documentation/Guide/Widgets/DataGrid/Enhance_Performance_on_Large_Datasets/#Rendering_Optimization) may help reduce rendering time and remove the gray boxes. In this mode, users can move to any page instantly.
```

Comment:
"If the rows take time" How would you quantify 'take time'?

Reply:
This depends on many things: local or remote data store, the presence of cell templates, the client's device. On old and slow devices it takes more time to render the widget.

Reply:
I guess there is a value that rules whether the gray boxes should be displayed, but we shouldn't mention it in the docs because it can be changed, and the docs will be outdated.

Reply:
How would you say?

Reply:
I would probably say something like "Row display gray boxes while they are loaded and rendered." This way it does not matter about how long it takes or which value determines it.

Reply:
If rendering is fast the DataGrid does not show gray boxes. The DataGrid has internal criterios to determine fast and slow rendering. So, "while they are loaded and rendered" is not correct.